packages/@sanity/form-builder/src/inputs/BlockEditor/Editor.js

```diff
@@ -150,16 +150,18 @@ export default class Editor extends React.Component<Props> {
         // Regular block
         this.scrollIntoView(change, block)
       }
-    } else if (focusPathChanged) {
+    } else if (focusPathChanged && !readOnly) {
       // Must be here to set focus after editing interfaces are closed
       change.focus()
       onChange(change)
     }
   }
 
   scrollIntoView(change, node) {
-    const {onChange} = this.props
-    change.collapseToEndOf(node)
+    const {onChange, readOnly} = this.props
+    if (!readOnly) {
+      change.collapseToEndOf(node)
+    }
     const element = findDOMNode(node)
     element.scrollIntoView({behavior: 'instant', block: 'center', inline: 'nearest'})
     onChange(change)
```

Comment:
I think this should not be called when readOnly. Though there probably aren't any changes, better be on the safe side.